pkg/raft/raft.go

```diff
@@ -1755,13 +1755,11 @@ func (r *raft) restore(s pb.Snapshot) bool {
 	if r.state != StateFollower {
 		// This is defense-in-depth: if the leader somehow ended up applying a
 		// snapshot, it could move into a new term without moving into a
-		// follower state. This should never fire, but if it did, we'd have
-		// prevented damage by returning early, so log only a loud warning.
+		// follower state. This should never fire, but if it did, we panic.
 		//
 		// At the time of writing, the instance is guaranteed to be in follower
 		// state when this method is called.
-		r.logger.Warningf("%x attempted to restore snapshot as leader; should never happen", r.id)
-		r.becomeFollower(r.Term+1, None)
+		r.logger.Panicf("%x attempted to restore snapshot as leader; should never happen", r.id)
```

Comment:
The condition above is `state != Follower`. Fix the message?

Is it true that this won't be called for `[Pre]Candidate` either? Is there something problematic with that case?

Reply:
I guess the snapshot is always sent on behalf of a leader@`Term` (is it true for delegated snapshots though?), so there is no sense in staying a (pre-)candidate at the same `Term`.

Reply:
All paths leading to here, make us `becomeFollower` first, so it checks out. I think it's sufficient to just fix the panic message here.